### PR TITLE
cryptsetup: support same_cpu_crypt and submit_from_crypt_cpus options

### LINKED ISSUE #11946: cryptsetup options for performance tuning are not available in crypttab; dm-crypt not usable this way
on my system, multithreaded dm-crypt is way slower than singlethreaded, only 24MiB/s write speed. 
dm-crypt and cryptsetup have documented parameters for this case (since 2015 i believe)

>   --perf-same_cpu_crypt
>   --perf-submit_from_crypt_cpus

see 
https://gitlab.com/cryptsetup/cryptsetup/commit/4f7b4136386aff41a1608727d0ca362e5589149d

with these parameters i can setup dm-crypt for full speed on my system 
there seems to be no way to use these with systemd crypttab

**systemd version the issue has been seen with**
current  master  ( i looked in the source) in cryptsetup-generator.c and cryptsetup.c
today last commit
https://github.com/systemd/systemd/commit/c20db3887569e0c0d9c0e2845c5286e7edf0133a

and ubuntu 18.04 systemd v 237  (i have it running on my system)

**Used distribution**

ubuntu 18.04 amd64

**Expected behaviour you didn't see**
be able to specify dm-crypt parameters in crypttab, alongside with "discard" parameter

> same_cpu_crypt
> submit_from_crypt_cpus

systemd would pass these to dm-crypt

**Unexpected behaviour you saw**
parameters get ignored in boot (warning in syslog)

**Steps to reproduce the problem**
setup crypttab this way (with valid uuid)

> $ sudo cat /etc/crypttab 
> sda5_crypt UUID=594f01cd-bfe1-400b-8af3-f770af9317b2 none luks,discard,same_cpu_crypt,submit_from_crypt_cpus

see warnings in syslog and check dm-crypt with command

> dmsetup table 

the parameters will not be listed, as they should next to "allow_discards"

### PR DESCRIPTION
**Pull request by yuwata**
Closes #11946.

### CODE REVIEW

**Comment by martinpitt** [comment]
This travis failure seems relevant:
```
../src/cryptsetup/cryptsetup.c: In function ‘run’:
../src/cryptsetup/cryptsetup.c:686:34: error: ‘CRYPT_ACTIVATE_SAME_CPU_CRYPT’ undeclared (first use in this function)
                         flags |= CRYPT_ACTIVATE_SAME_CPU_CRYPT;
                                  ^
../src/cryptsetup/cryptsetup.c:686:34: note: each undeclared identifier is reported only once for each function it appears in
../src/cryptsetup/cryptsetup.c:689:34: error: ‘CRYPT_ACTIVATE_SUBMIT_FROM_CRYPT_CPUS’ undeclared (first use in this function)
                         flags |= CRYPT_ACTIVATE_SUBMIT_FROM_CRYPT_CPUS;
                                  ^
```
Presumably Ubuntu 16.04 has a too old kernel for this.

**Comment by yuwata** [comment]
Thanks. Updated.

File: src/cryptsetup/cryptsetup.c [hunk 1]
```diff
@@ -199,6 +201,10 @@ static int parse_one_option(const char *option) {
                 arg_verify = true;
         else if (STR_IN_SET(option, "allow-discards", "discard"))
                 arg_discards = true;
+        else if (streq(option, "same_cpu_crypt"))
+                arg_same_cpu_crypt = true;
+        else if (streq(option, "submit_from_crypt_cpus"))
+                arg_submit_from_crypt_cpus = true;
```
Comment (poettering) [inline src/cryptsetup/cryptsetup.c:207]: hmm, all other options we use here (much like the options in fstab) use dashes rather underscores for separating words. I think these ones should do too. (or maybe we should do it like on the kernel cmdline, and consider _ and - the same, i.e. generalize the code in src/basic/proc-cmdline.c a bit, and maybe add streq_dash_underscore_same() or so which is like streq() but considers dash and underscore the same?

Comment (yuwata) [reply src/cryptsetup/cryptsetup.c:207]: Indeed...

Comment (yuwata) [reply src/cryptsetup/cryptsetup.c:207]: Hmm... Complicated...

**Comment by yuwata** [comment]
Updated. Now the option is named with dashes. I noticed that our options are slightly different from `dm-crypt` used https://github.com/torvalds/linux/blob/master/Documentation/device-mapper/dm-crypt.txt. So, previous underscored names are dropped.

**Pull request merged**